crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs

```diff
@@ -71,6 +71,63 @@ impl Violation for Airflow3Removal {
     }
 }
 
+const REMOVED_CONTEXT_KEYS: [&str; 12] = [
+    "conf",
+    "execution_date",
+    "next_ds",
+    "next_ds_nodash",
+    "next_execution_date",
+    "prev_ds",
+    "prev_ds_nodash",
+    "prev_execution_date",
+    "prev_execution_date_success",
+    "tomorrow_ds",
+    "yesterday_ds",
+    "yesterday_ds_nodash",
+];
+
+fn extract_name_from_slice(slice: &Expr) -> Option<String> {
+    match slice {
+        Expr::StringLiteral(ExprStringLiteral { value, .. }) => Some(value.to_string()),
+        _ => None,
+    }
+}
+
+pub(crate) fn removed_context_variable(checker: &mut Checker, expr: &Expr) {
+    if let Expr::Subscript(ExprSubscript { value, slice, .. }) = expr {
+        if let Expr::Name(ExprName { id, .. }) = &**value {
+            if id.as_str() == "context" {
```

Comment:
Is it possible to support code like this?

```python
c = get_current_context()
c["execution_date"]
```

We should probably not hard code the variable name.

Reply:
I think we'll need to check 2 cases here.

1. the `context` is from a function argument (e.g., `def func(**context):`)
2. the variable is assigned from `get_current_context`. we probably could do something like https://github.com/astral-sh/ruff/blob/bec8441cf5c0e3870f7e21cc042937d4279268bd/crates/ruff_linter/src/rules/airflow/rules/removal_in_3.rs#L265-L267

Reply:
Added the check for both.